### ISSUE TITLE
defaultPath option to showSaveDialog() has no effect on macOS

### ISSUE THREAD

**Issue by cbfiddle** [electron/electron#6306]
* Electron version: 1.2.5
* Operating system: 10.10.5

It does not affect the directory that is initially displayed. The file name field starts out empty.

**Comment by zcbenz** [comment]
Can you show how you are calling the API? It works for me.

**Comment by zcbenz** [comment]
Closing until more information is provided.

**Comment by cbfiddle** [comment]
I believe I have found the cause for this problem, which was a mistake on my part. I passed an incorrect argument for the browser window optional parameter, and because it was not a browser window, it was interpreted as options. (The second parameter, which was the options, must have been taken as the optional callback parameter.)

Could there be some additional error checking on the argument list? This is an easy mistake to make in Javascript with no static type checking.

After fixing that problem, I observe one more issue:

If the default path names a package (a directory tagged in FinderInfo to be displayed as a file), it is interpreted as a directory, which means that the directory is displayed in the combo box and the file name field is initialized with Untitled. This is not the correct behavior, as one can observe in TextEdit with an rtfd format document.
